tracing-appender/src/writer.rs

```diff
@@ -0,0 +1,115 @@
+use crate::rolling::create_writer_file;
+use std::fs::File;
+use std::io;
+
+#[cfg(feature = "compression_gzip")]
+use {
+    crate::compression::CompressionConfig,
+    flate2::write::GzEncoder,
+    std::io::BufWriter,
+    std::sync::{Arc, RwLock},
+};
+
+#[cfg(feature = "compression_gzip")]
+#[derive(Debug)]
+pub(crate) struct CompressedGzip {
+    compression: CompressionConfig,
+    buffer: Arc<RwLock<BufWriter<GzEncoder<BufWriter<File>>>>>,
+}
+
+#[derive(Debug)]
+pub(crate) enum WriterChannel {
+    File(File),
+    #[cfg(feature = "compression_gzip")]
+    CompressedFileGzip(CompressedGzip),
+}
+
+impl WriterChannel {
+    #[cfg(feature = "compression_gzip")]
+    pub(crate) fn new(
+        directory: &str,
+        filename: &str,
+        #[cfg(feature = "compression_gzip")] compression: Option<CompressionConfig>,
```

Comment:
nit: since the entire function is feature-flagged, i think this argument doesn't need its own `#[cfg]` attribute:
```suggestion
        compression: Option<CompressionConfig>,
```